api/test/opentelemetry/distributed_context/propagation/text_formatter_shared_test_cases.rb

```diff
@@ -0,0 +1,112 @@
+# frozen_string_literal: true
+
+# Copyright 2019 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module TextFormatterSharedTestCases
+  def self.included(base)
+    base.class_eval(&TESTS)
```

Comment:
With Minitest (classic) sharing test cases is straight forward and is as simple as including a module with the shared test cases. Sharing tests with Minitest Spec using module inclusion proved to be much more of a challenge. It works if test cases are all defined within `describe` blocks, but anything more advanced becomes progressively more difficult. `let` cannot exist outside of `describe` and sharing state, such as variables defined with `let`, from the including class are issues that I was running into. This approach, while ugly, solves them. If anyone knows of a better way to share test cases and state from the including class with Minitest Spec, let me know.